File: DynamicProgramming/BricksGame/BricksGame.java
import java.util.List;

public class BricksGame {
    public static int bricksGame(List<Integer> arr) {
        int total = 0;
        int[] dp = new int[arr.size()];
        for(int i = arr.size()-1, temp = 0; i >= 0; i--, temp++){
            total += arr.get(i);
            if (temp < 3){
                dp[i] = (int)total;
            }
            else{
                dp[i] = (int)total - Math.min(Math.min(dp[i+1], dp[i+2]), dp[i+3]);
            }
        }
        return dp[0];
    }
}

File: DynamicProgramming/BricksGame/BricksGame.py
def bricksGame(arr):
    dp = [0]*len(arr)
    tot = 0
    temp = 0
    for i in range(len(arr)-1, -1, -1):
        tot += arr[i]
        if (temp < 3):
            dp[i] = tot
            temp += 1
        else:
            dp[i] = tot - min(dp[i+1], dp[i+2], dp[i+3])
        
    return dp[0]